q2_diversity/_alpha/alpha_rarefaction_assets/src/toolbar.js

```diff
@@ -0,0 +1,40 @@
+/* global d */
+import state from './state';
+
+export function addMetricPicker(row, metrics, selectedMetric) {
+  const grp = row.append('div').attr('class', 'col-lg-2 form-group metricPicker');
+  grp.append('label').text('Metric');
+  grp.append('select')
```

Comment:
Then you are just "hydrating" the nodes with the `.data` instead of creating the dom, which makes it easier to hot-swap the contents when `.data` changes using the same function because you'd be working with selectors instead of mutating the DOM (if that were to ever be needed).